exercises/zebra-puzzle/.meta/src/reference/java/ZebraPuzzle.java

```diff
@@ -0,0 +1,156 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+class ZebraPuzzle {
+    private String waterDrinker = "";
+    private String zebraOwner = "";
+
+    ZebraPuzzle() {
+        this.solve();
+    }
+
+    private static boolean isJustRightOf(int a, int b) {
+        return a - 1 == b;
+    }
+
+    private static boolean nextTo(int a, int b) {
+        return isJustRightOf(a, b) || isJustRightOf(b, a);
+    }
+
+    private void solve() {
+        int[] houses = new int[]{1, 2, 3, 4, 5};
+        int first = 1;
+        int middle = 3;
+
+        perms(houses).forEach(colors -> {
```

Comment:
Since this is iterating through the permutations maybe `colorPermutation` might be a more descriptive name? 🙂 Same for the other iterations through permutations 🙂